### PR TITLE
fix 1 replica balance failed, rebuild index ut

### PR DESCRIPTION
**Pull request by critical27**
Fixes https://github.com/vesoft-inc/nebula-graph/issues/969

### CODE REVIEW

File: src/storage/admin/AdminProcessor.h [hunk 1]
```diff
@@ -58,7 +58,6 @@ class TransLeaderProcessor : public BaseProcessor<cpp2::AdminExecResp> {
         if (partMeta.hosts_.size() == 1) {
             LOG(INFO) << "Skip transfer leader of space " << spaceId
                       << ", part " << partId << " because of single replica.";
-            this->pushResultCode(cpp2::ErrorCode::E_TRANSFER_LEADER_FAILED, partId);
```
Comment (randomJoe211) [inline src/storage/admin/AdminProcessor.h:61]: Why remove the error msg? The user should know if the BALANCE operation worked. And if the operation failed, we should provide a clear cause like "To balance the xx, the graph spaces must have more than one replicas."

File: src/storage/test/RebuildIndexTest.cpp [hunk 2]
```diff
@@ -406,8 +406,7 @@ TEST_F(RebuildIndexTest, RebuildEdgeIndexWithDelete) {
         auto* processor = DeleteEdgesProcessor::instance(RebuildIndexTest::env_, nullptr);
         auto fut = processor->getFuture();
         processor->process(req);
-        auto resp = std::move(fut).get();
-        EXPECT_EQ(0, resp.result.failed_parts.size());
```
Comment (bright-starry-sky) [inline src/storage/test/RebuildIndexTest.cpp:410]: A good way.

File: src/storage/admin/AdminProcessor.h [hunk 3]
```diff
@@ -58,7 +58,6 @@ class TransLeaderProcessor : public BaseProcessor<cpp2::AdminExecResp> {
         if (partMeta.hosts_.size() == 1) {
             LOG(INFO) << "Skip transfer leader of space " << spaceId
```
Comment (darionyaphet) [inline src/storage/admin/AdminProcessor.h:59]: It seems not allowed to balance for the single replica cluster. 

Should we support some warning before the balance be executed?

Comment (critical27) [reply src/storage/admin/AdminProcessor.h:59]: Why forbid it, it is legal

File: src/storage/admin/AdminProcessor.h (hunk 1 above)
Comment (critical27) [reply src/storage/admin/AdminProcessor.h:61]: This is way more complicated than you thought. Actually, there won't be this phase for 1 replica balance

Comment (darionyaphet) [reply src/storage/admin/AdminProcessor.h:61]: the single replica may not require transfer leader so it's fine @randomJoe211

**Review by bright-starry-sky — APPROVED** [review]
LGTM

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**